feature_engine/discretisation/base_discretiser.py

```diff
@@ -0,0 +1,120 @@
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.dataframe_checks import _check_contains_na
+
+import warnings
+
+
+class BaseDiscretiser(BaseNumericalTransformer):
+    """
+    Share set-up checks and methods across numerical discretizers
+
+    Parameters
+    ----------
+    return_object: bool, default=False
+        Whether the the discrete variable should be returned as numeric or as
+        object. If you would like to proceed with the engineering of the variable as if
+        it was categorical, use True. Alternatively, keep the default to False.
+
+    return_boundaries : bool, default=False
+        Whether the output should be the interval boundaries. If True, it returns
+        the interval boundaries. If False, it returns integers.
+
+
+    """
+
+    def __init__(
+            self,
+            return_object: bool = False,
+            return_boundaries: bool = False,
+    ) -> None:
+
+        if not isinstance(return_object, bool):
+            raise ValueError("return_object must be True or False")
+
+        if not isinstance(return_boundaries, bool):
+            raise ValueError("return_boundaries must be True or False")
+
+        self.return_object = return_object
+        self.return_boundaries = return_boundaries
+
+    def _check_transform_input_and_state(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Checks that the input is a dataframe and of the same size than the one used
+        in the fit method. Checks absence of NA.
+
+        Parameters
+        ----------
+        X: Pandas DataFrame
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame
+        ValueError
+            - If the variable(s) contain null values.
+            - If the df has different number of features than the df used in fit()
+
+        Returns
+        -------
+        X: Pandas DataFrame
+            The same dataframe entered by the user.
+        """
+
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check that input is a dataframe
+        X = _is_dataframe(X)
+
+        # Check input data contains same number of columns as df used to fit
+        _check_input_matches_training_df(X, self.n_features_in_)
+
+        # check if dataset contains na
+        _check_contains_na(X, self.variables_)
+
+        return X
+
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """Replace categories with the learned parameters.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features].
+            The dataset to transform.
+
+        Returns
+        -------
+        X_new: pandas dataframe of shape = [n_samples, n_features].
+            The dataframe containing the categories replaced by numbers.
+        """
+
+        X = self._check_transform_input_and_state(X)
```

Comment:
we are inheriting the transform method of the BaseNumericalTransformer, so instead of this, we can use:

`X = super().transform(X)`

That will do the checks you want.